### ISSUE TITLE
+Closeup Zoom: doesn't work for 3D Weapon Icons

### ISSUE THREAD

**Issue by DahBlount** [scp-fs2open/fs2open.github.com#601]
+Closeup Zoom: only appears to work for the spinning 3D Models in the weapon selection screen, not the icons. In the code it appears to attempt setting the fov for the icon but it isn't successful in doing so. Relevant code is in missionweaponchoice.cpp, lines 2751 and 2992. Function is called draw_model_icon().

**Comment by niffiwan** [comment]
Just checking- you mean it doesn't work for weapons, but it does work for ships in ship selection?

Also; vaguely related, IMHO +Closeup Zoom: affects too many things, it may be useful to have different controls for icons, rotating 3D model, HUD targetview, etc.

**Comment by DahBlount** [comment]
No. When in the weapon selection screen, there are weapon icons on the bottom left that you can drag and when you click on an icon it opens up a rotating 3D model on the top right. +Closeup Zoom: work for the rotating model, but not the icon. I agree that +Closeup Zoom: is used for too many separate features. Perhaps we should look into separating these in the future?

**Comment by MageKing17** [comment]
I have to wonder if the time to separate them out isn't right now, since the icon part of the feature isn't working, so no existing mods can be relying on it affecting the icons.

**Comment by niffiwan** [comment]
Where is Closeup_pos & Closeup_zoom used?

- HUD: autogenerated shield icons (zoom is *2.5)
- HUD: hardpoints gauge (zoom is *1.5)
- HUD: target box
- F3 Lab
- Techroom (zoom is *1.3) 
- Techroom shadows (but may not be normal usage; sets various distances from z-value?)
- Briefing icon closeup view (check this further)
- Briefing draw_model_icon (zoom is *2.5) (check this further)
- Briefing Ship Select (zoom *1.3)
- Briefing Weapon choice wl_render_overhead_view (zoom *1.3)
- Briefing Weapon choice weapon_select_do (zoom *0.65)
- Scripting renderTechModel & renderTechModel2 (zoom defaults to *1.3)
